### PR TITLE
Add support in `+lang/ocaml` for `dune`.

### PR DESCRIPTION
**Pull request by lyguy**
Dune is a popular build-tool for OCaml projects, the `dune` package adds:
 - Adds syntax highlighting
 - Template insertion
 - Some testing/compilation functions

### CODE REVIEW

File: layers/+lang/ocaml/packages.el [hunk 1]
```diff
@@ -32,6 +33,34 @@
       :modes merlin-mode
       :variables merlin-completion-with-doc t)))
 
+(defun ocaml/init-dune ()
```
Comment (smile13241324) [inline layers/+lang/ocaml/packages.el]: Please add code to load `dune-mode` after opening a dune file.
For an example please have a look at the package.el of `ietf` layer.

File: layers/+lang/ocaml/packages.el [hunk 2]
```diff
@@ -32,6 +33,34 @@
...
+        "ii" 'dune-insert-install-form
+        "ic" 'dune-insert-copyfiles-form
+        "it" 'dune-insert-tests-form
+        "iv" 'dune-insert-env-form
+        "id" 'dune-insert-ignored-subdirs-form
+        "tP" 'dune-promote
+        "tp" 'dune-runtest-and-promote
+        "cc" 'compile)
+      (spacemacs/declare-prefix-for-mode 'dune "mc" "compile/check")
+      (spacemacs/declare-prefix-for-mode 'dune "mi" "insert-form"))))
```
Comment (smile13241324) [inline layers/+lang/ocaml/packages.el]: I think the prefix for `t` is missing.

Comment (lyguy) [reply layers/+lang/ocaml/packages.el]: Yep. Added declarations for both `dune-mode` and `tuareg-mode.

File: layers/+lang/ocaml/packages.el (hunk 1 above)
Comment (lyguy) [reply layers/+lang/ocaml/packages.el]: Added.

**Comment by lyguy** [comment]
Hi @smile13241324,
Thanks for the review. I've pushed fixes which address you comments.

CI failed on `spacemacs-base distribution functional tests`, with the error:
```
...
(Spacemacs) --> installing package: overseer@emacs-lisp... [16/16]
Cannot open load file: No such file or directory, bind-map
../../../../spacemacs.mk:39: recipe for target 'func_tests' failed
```

Since the changes are all to `+lang/ocaml` this _feels_ like a melpa issue (a la #8984, #6767), but I'm not sure. What do you think?

**Comment by lyguy** [comment]
Hi @smile13241324,

Thanks for the catch. I've fixed the prefixes for `dune-mode`.

Regarding `dune-project` files, they're a part of dune projects, but `dune-mode` doesn't have support there yet.  I copied the [auto-load regex](https://github.com/lyguy/spacemacs/blob/ocaml-dune/layers/%2Blang/ocaml/packages.el#L66) in `packages.el` verbatim from [`dune.el`](https://github.com/ocaml/dune/blob/master/editor-integration/emacs/dune.el#L386), so I'd vote to leave it as is.

Thanks!

**Comment by smile13241324** [comment]
Thank you for contributing to Spacemacs!
The PR has been cherry-picked into the develop branch, you can safely delete your branch.

**Pull request closed without merging**